arangod/Transaction/Manager.cpp

```diff
@@ -785,10 +785,15 @@ std::shared_ptr<transaction::Context> Manager::leaseManagedTrx(
   TRI_IF_FAILURE("leaseManagedTrxFail") { return nullptr; }
 
   auto role = ServerState::instance()->getRole();
+  std::chrono::steady_clock::time_point now = std::chrono::steady_clock::now();
   std::chrono::steady_clock::time_point endTime;
   if (!ServerState::isDBServer(role)) {  // keep end time as small as possible
-    endTime = std::chrono::steady_clock::now() +
-              std::chrono::milliseconds(int64_t(1000 * _streamingLockTimeout));
+    endTime =
+        now + std::chrono::milliseconds(int64_t(1000 * _streamingLockTimeout));
+  }
+  std::chrono::steady_clock::time_point detachTime;
+  if (_streamingLockTimeout >= 1.0) {
+    detachTime = now + std::chrono::milliseconds(1000);
   }
```

Comment:
Right now the detaching is not restricted to the instance types single server or DB server.
Do we want it to happen on coordinators as well, or just single servers and DB servers?

Reply:
I think we want detaching on all types of servers, even on agents. It is a security measure and could potentially be relevant everywhere.